Source/background.js

```diff
@@ -76,31 +60,111 @@ function translateOneRequestPerFewWords(words, prefs, callback) {
     translateORPFWRec(concatWordsArray, 1, cWALength, tMap, prefs, callback);
 }
 
+var translator = {
+    translate: function(prefs, word) {}
+};
+
+var googleTranslator = function() {};
+googleTranslator.prototype = Object.create(translator);
+
+var yandexTranslator = function() {};
+yandexTranslator.prototype = Object.create(translator);
+
+/**
+ * @desc Constructs google translate url
+ * @params User saved preferences
+ * @params word to be translated
+ * @return Google Translate query from source to destination language
+ */
+googleTranslator.prototype.translate = function(prefs, word, callback){
+    var url = 'http://translate.google.com/translate_a/t?client=f&otf=1&pc=0&hl=en';
+    var sL = prefs["sourceLanguage"];
+    if(sL != 'auto') {
+        url += '&sl=' + sL;
+    }
+    url += '&tl=' + prefs["targetLanguage"];
+    url += '&text=';
+    url += word;
+
+    getData(url, function(result){
+        var tMap = {}
+        for (i in result.sentences) {
+            var orig = result.sentences[i].orig;
+            var origT = orig.substring(0,orig.length - 1);
+            var trans = result.sentences[i].trans;
+            var transT = trans.replace(/[.\u3002]/,""); // removes punctuation
+            tMap[origT] = transT;
+        }
+        callback(tMap);
+    });
+}
+
+yandexTranslator.prototype.translate = function(prefs, word, callback){
+    var apikey = "trnsl.1.1.20150429T202526Z.d63a1483e719555a.852f432bad29a4628c1d1694b6114c0efb22f183"
```

Comment:
Yes, i should be able to get it done today, also i was thinking of moving the Translation Service option to "Create New Translation Configuration", so a user can select different translators for different languages. We can then limit the set of language which we show to the ones which translator supports. What do you think about this design?